### ISSUE TITLE
EnvironmentAWSCredentialsProvider causes log spam

### ISSUE THREAD

**Issue by chancez** [aws/aws-sdk-cpp#2533]
### Describe the bug

See https://github.com/ClickHouse/ClickHouse/issues/41720.

TL;DR: we're seeing hundreds/thousands of logs per second spamming:
```
2022.09.23 09:23:30.838551 [ 296 ] {} [Information](https://app.slack.com/client/T40ANG0TH/threads/thread/Information) AWSClient: EnvironmentAWSCredentialsProvider: Found secret key
```

### Expected Behavior

Less logs. Maybe once at startup. Environment variables cannot be changed for a running application, so caching and reusing this environment variable should suffice. No need to look it up every call, and especially not necessary to log every time.

### Current Behavior

Logs get produced ~every time a request is made that needs credentials

### Reproduction Steps

This problem exists in a downstream project, so I do not have a minimal working example, but the issue should be incredibly easy to reproduce.

### Possible Solution

Cache the values of the env-vars in the EnvironmentAWSCredentialsProvider indefinitely.

### AWS CPP SDK version used

1.11.98

### Compiler and Version used

na

### Operating System and version

Linux

**Comment by yasminetalby** [comment]
Hello @chancez ,

Thank you very much for your submission. 
This behavior is expected. Logs get produced every time you are retrieving credentials through the EnvironmentAWSCredentialsProvider ([see](https://github.com/aws/aws-sdk-cpp/blob/63d037e173dbca3f3017c4a9ea565511858f9713/src/aws-cpp-sdk-core/source/auth/AWSCredentialsProvider.cpp#L71)).  
I do understand the disturbance you experience. I'll bring this topic up to discussion with the team and will update you here.

We appreciate your feedback and collaboration.

Best regards,

Yasmine
